internal/resolver/dns/dns_resolver.go

```diff
@@ -86,24 +86,24 @@ var (
 	minDNSResRate = 30 * time.Second
 )
 
-var customAuthorityDialler = func(authority string) func(ctx context.Context, network, address string) (net.Conn, error) {
+var customHostDialler = func(urlHost string) func(ctx context.Context, network, address string) (net.Conn, error) {
 	return func(ctx context.Context, network, address string) (net.Conn, error) {
```

Comment:
Since `address` is used, it might be better to replace with `_` to indicate that more readily.